src/campaign_server/addon_utils.cpp

```diff
@@ -92,18 +92,14 @@ std::string format_addon_feedback_url(const std::string& format, const config& p
 	return std::string();
 }
 
-void support_translation(config& addon, std::string locale_id)
+void support_translation(config& addon, const std::string locale_id)
 {
-	if(!locale_id.empty()) {
-		config& locale = addon.find_child("translation", "language", locale_id);
-		if(locale) {
-			locale["supported"] = true;
-		}
-		else {
-			addon.add_child("translation")["language"] = locale_id; //Doing this to circumvent a weird validation bug
-			addon.find_child("translation", "language", locale_id)["supported"] = true;
-		}
+	config* locale = &addon.find_child("translation", "language", locale_id);
+	if(!*locale) {
+		locale = &addon.add_child("translation");
+		(*locale)["language"] = locale_id;
 	}
+	(*locale)["supported"] = true;
```

Comment:
Ok, I made `locale_id` a const reference. However, I left the pointer untouched in this place because the reference reassignment had caused the server crash.

<details>
  <summary>The old crashing code</summary>

```
void support_translation(config& addon, const std::string& locale_id)
{
	config* locale = &addon.find_child("translation", "language", locale_id);
	if(!*locale) {
		locale = &addon.add_child("translation");
		(*locale)["language"] = locale_id;
	}
	(*locale)["supported"] = true;
}
```

</details>